docs/docs/guides/labs/components/creating-new-component-types/testing-your-component-type.md

```diff
@@ -2,9 +2,106 @@
 description: How to test component types.
 sidebar_position: 500
 title: Testing your component type
-unlisted: true
 ---
 
 import DgComponentsPreview from '@site/docs/partials/\_DgComponentsPreview.md';
 
 <DgComponentsPreview />
+
+## Testing custom components
+
+Component authors need to test their components in various contexts. The Dagster framework provides utilities in order to faciliate this process
+
+### The core workhorse: `scaffold_defs_sandbox`
+
+The function at the core of our testing workflows is `scaffold_defs_sandbox.`
+
+The function signature is the following:
+
+```python
+@contextmanager
+def scaffold_defs_sandbox(
+    *,
+    component_cls: type,
+    scaffold_params: Optional[dict[str, Any]] = None,
+    component_path: Optional[Union[Path, str]] = None,
+    scaffold_format: ScaffoldFormatOptions = "yaml",
+    project_name: Optional[str] = None,
+) -> Iterator[DefsPathSandbox]: ...
+```
+
+For the purposes of this guide we will only concern ourselves with `component_cls` and `scaffold_params`. Users are highly unlikely to require the other parameters.
```

Comment:
```suggestion
For the purposes of this guide, we will only concern ourselves with `component_cls` and `scaffold_params`, since you are highly unlikely to require the other parameters.
```